integration-tests/byoid/integration_test.go

```diff
@@ -231,3 +240,84 @@ func TestURLBasedCredentials(t *testing.T) {
 		},
 	})
 }
+
+// Tests to make sure AWS based external credentials work properly.
+func TestAWSBasedCredentials(t *testing.T) {
+	data := neturl.Values{}
+	data.Set("audience", clientID)
+	data.Set("includeEmail", "true")
+
+	client, err := google.DefaultClient(context.Background(), "https://www.googleapis.com/auth/cloud-platform")
+	resp, err := client.PostForm(fmt.Sprintf("https://iamcredentials.googleapis.com/v1/%s:generateIdToken", clientID), data)
+	if resp.StatusCode != 200 {
+		t.Errorf("byoid: failed to get Google ID token for AWS test: %v", err)
```

Comment:
This prefix convention is only used in non-test code when returning errors. Please see this example for good testing errors: https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures